src/typeInference/annotateFlworExpression.ts

```diff
@@ -48,17 +48,25 @@ export function annotateFlworExpression(
 					annotationContext,
 					annotate
 				);
-				if (hasFor) {
+
+				if (!retType) {
 					retType = { type: ValueType.ITEM, mult: SequenceMultiplicity.ZERO_OR_MORE };
+				} else if (hasFor) {
+					retType = { type: retType.type, mult: SequenceMultiplicity.ZERO_OR_MORE };
 				}
+
```

Comment:
retType should only be inserted when it's not an item()*, so the insertAttribute() call should be moved into the else if

Reply:
This was indeed not correct this has been fixed now